### PR TITLE
Do not call next when the carousel or the parent isn't visible

### PR DESCRIPTION
**Pull request by Johann-S**
- [x] Unit test to be sure the carousel don't cycle when we cannot see it
- [x] Unit test to be sure the carousel don't cycle when is parent isn't visible

~Work in progress...~

Close : #23523

### CODE REVIEW

File: js/src/carousel.js [hunk 1]
```diff
@@ -131,7 +131,9 @@ const Carousel = (($) => {
 
     nextWhenVisible() {
       // Don't call next when the page isn't visible
-      if (!document.hidden) {
+      // or the carousel or it's parent isn't visible
+      if (!document.hidden &&
+        ($(this._element).is(':visible') && $(this._element).css('visibility') !== 'hidden')) {
```
Comment (XhmikosR) [inline js/src/carousel.js]: Are you sure those 2 checks aren't identical?

Comment (Johann-S) [inline js/src/carousel.js]: Yes the first one just check if the parent or the carousel don't have `display: none;`

Comment (Johann-S) [inline js/src/carousel.js]: For more informations and precise explanations (with a better English 😆) : https://api.jquery.com/hidden-selector/

**Review by XhmikosR — APPROVED** [review]

**Comment by XhmikosR** [comment]
So we just need a test with both cases and we are all set. Thanks for the fast patch!

**Comment by Johann-S** [comment]
I added the two unit tests so @XhmikosR if you're available for a quick code review 😄

File: js/tests/unit/carousel.js [hunk 2]
```diff
@@ -879,4 +879,43 @@ $(function () {
...
+    $html.appendTo('#qunit-fixture')
+    var $parent = $html.find('#parent')
+    var $carousel = $html.find('#myCarousel')
+    $carousel.bootstrapCarousel()
+    var $firstItem = $('#firstItem')
+
+    setTimeout(function () {
+      assert.ok($firstItem.hasClass('active'))
+      $carousel.bootstrapCarousel('dispose')
+      $parent.attr('style', 'visibility: hidden;')
```
Comment (XhmikosR) [inline js/tests/unit/carousel.js]: This add `visibility` but doesn't remove `display:none`, right?

Comment (Johann-S) [inline js/tests/unit/carousel.js]: It remove the `display: none;` because we set the attribute value no matter what we had before that

Comment (XhmikosR) [inline js/tests/unit/carousel.js]: OK, looks good to me then. If you can't think of anything else, please merge :)

**Pull request merged**